raiden/network/transport/matrix/client.py

```diff
@@ -501,18 +514,20 @@ def _sync(self, timeout_ms: int, latency_ms: int) -> None:
             user_id=self.user_id,
             sync_iteration=self.sync_iteration,
             sync_filter_id=self._sync_filter_id,
-            last_sync_time=self.last_sync,
+            last_sync_time=self.last_sync_returned,
         )
 
         time_before_sync = time.monotonic()
-        time_since_last_sync_in_seconds = time_before_sync - self.last_sync
+        time_between_syncs_in_seconds = time_before_sync - self.last_sync_returned
 
-        # If it takes longer than `timeout_ms + latency_ms` to call `_sync`
-        # again, we throw an exception.  The exception is only thrown when in
-        # development mode.
+        # A matrix user presence changes to offline if the time *between* syncs
+        # exceeds a presence timeout (This is a synapse configuration, defaults
+        # to 30seconds). This has caused routing problems in the past, so, for
+        # the development mode the elapsed time is measured, and if it is too
+        # high an exception is raised.
         timeout_in_seconds = (timeout_ms + latency_ms) // 1_000
         timeout_reached = (
-            time_since_last_sync_in_seconds >= timeout_in_seconds
+            time_between_syncs_in_seconds >= timeout_in_seconds
```

Comment:
This seems like a silly check now, since the `_sync` does *very little* work. However, this lead to many bugs in the past, we decided to keep it in. (e.g. on startup the node will be cpu bound, decoding all the events from the blockchain can lead to starvation, in theory it is possible for a cpu hungry task to be scheduled before the next /sync is performed, the check here detects that in development mode)